src/api/courses/api.ts

```diff
@@ -0,0 +1,12 @@
+import privateApi from "src/configs/private-api";
+import type { ICourses, ICourseWithLessons } from "./types";
+
+export const getCourses = async (): Promise<ICourses> => {
+  return privateApi.get("/core/preview-courses").then((res) => res.data);
+};
```

Comment:
Initially, we got **unhandled** error 401 and also Axios returns "ERR_BAD_REQUEST"